task_router_workflow.go

```diff
@@ -0,0 +1,80 @@
+package twilio
+
+import (
+	"context"
+	"net/url"
+)
+
+const WorkflowPathPart = "Workflows"
+
+type WorkflowService struct {
+	client       *Client
+	workspaceSid string
+}
+
+type Workflow struct {
+	Sid                           string     `json:"sid"`
+	AccountSid                    string     `json:"account_sid"`
+	FriendlyName                  string     `json:"friendly_name"`
+	Configuration                 string     `json:"configuration"`
```

Comment:
I'm wondering if we should make this a `json.RawMessage`. Can you add a test that declares a struct and tries to unmarshal `Configuration` into it?

At the very least it should probably be a `[]byte` because that's what `json.Unmarshal` accepts and that's the most likely thing people are going to do with this field.

Reply:
Or something more like this?

```
func TestDecodeWorkflow(t *testing.T) {
	t.Parallel()
	msg := new(Workflow)
	if err := json.Unmarshal(testdata.WorkflowResponse, &msg); err != nil {
		t.Fatal(err)
	}
	if msg.Sid != "WW63868a235fc1cf3987e6a2b67346273f" {
		t.Errorf("wrong sid")
	}

       // How to encode json.RawMessage here?
}
```

Reply:
Basically we should be able to get a `Workflow` object, and then be able to Unmarshal the Configuration field separately.

```go
msg := new(Workflow)
json.Unmarshal(resp, msg)
cfg := new(TestConfig) // declare this struct in the test file
json.Unmarshal(msg.Configuration, cfg)
// make assertions about config
```

Reply:
How can I build a `json.RawMessage`?

```
func TestDecodeWorkflow(t *testing.T) {
	t.Parallel()
	msg := new(Workflow)

	if err := json.Unmarshal(testdata.WorkflowResponse, &msg); err != nil {
		t.Fatal(err)
	}

	cfg := Workflow{Configuration: ""} // How can I encode a string here?
	json.Unmarshal(msg.Configuration, cfg)
	if err := json.Unmarshal(cfg.Configuration, &msg); err != nil {
		t.Fatal(err)
	}
}
```

Reply:
Something like this? Obv need to pull stuff out into test file..

```
func TestDecodeWorkflow(t *testing.T) {
	t.Parallel()
	msg := new(Workflow)
	configuration := []byte(`{"task_routing":{"default_filter":{"queue":"WQ0c1274082082355320d8a41f94eb57aa"}}}`)

	if err := json.Unmarshal(testdata.WorkflowResponse, &msg); err != nil {
		t.Fatal(err)
	}

	cfg := Workflow{Sid: "", Configuration: configuration}
	if err := json.Unmarshal(cfg.Configuration, &msg); err != nil {
		t.Fatal(err)
	}
}
```

Reply:
`testdata.WorkflowResponse` already has the configuration on it - you shouldn't need to specify it again. Did you try the sample code that I posted? It should just work - you should be able to call Unmarshal with a `json.RawMessage` as the first argument.

Reply:
I've added the test (https://github.com/kevinburke/twilio-go/pull/33/commits/75c5534873c085bbddf3ad0fa4f7feea35164249), feel free to make any changes to improve it.